linkerd/app/integration/src/proxy.rs

```diff
@@ -147,41 +147,43 @@ struct DstInner {
     outbound_local_addr: Option<SocketAddr>,
 }
 
-impl app::transport::GetOriginalDst for MockOriginalDst {
-    fn get_original_dst(&self, sock: &dyn app::transport::AddrInfo) -> Option<SocketAddr> {
-        sock.local_addr().ok().and_then(|local| {
-            let inner = self.0.lock().unwrap();
-            if inner.inbound_local_addr == Some(local) {
-                inner.inbound_orig_addr
-            } else if inner.outbound_local_addr == Some(local) {
-                inner.outbound_orig_addr
-            } else {
-                None
-            }
+impl app::core::transport::OrigDstAddr for MockOriginalDst {
+    fn orig_dst_addr(&self, sock: &tokio::net::TcpStream) -> Option<SocketAddr> {
+        info_span!("mock original dst").in_scope(|| {
+            sock.local_addr().ok().and_then(|local| {
+                let inner = self.0.lock().unwrap();
+                if inner.inbound_local_addr == Some(local) {
+                    debug!(local = %local, mock = ?inner.inbound_orig_addr, "inbound");
+                    inner.inbound_orig_addr
+                } else if inner.outbound_local_addr == Some(local) {
+                    debug!(local = %local, mock = ?inner.outbound_orig_addr, "outbound");
+                    inner.outbound_orig_addr
+                } else {
+                    debug!(local = %local, "failed");
+                    None
+                }
+            })
         })
     }
 }
 
-fn run(proxy: Proxy, mut env: app::config::TestEnv) -> Listening {
+fn run(proxy: Proxy, mut env: TestEnv) -> Listening {
     let controller = proxy.controller.unwrap_or_else(|| controller::new().run());
     let inbound = proxy.inbound;
     let outbound = proxy.outbound;
     let identity = proxy.identity;
     let mut mock_orig_dst = DstInner::default();
 
     env.put(
-        app::config::ENV_DESTINATION_SVC_ADDR,
+        "LINKERD2_PROXY_DESTINATION_SVC_ADDR",
```

Comment:
nit: it would be nice if this could still use a const...i won't block the PR over that, though...

Reply:
this const existed only for this usage, and was redundant/decoupled with the templatized consts that actually control how configs are parsed. in light of all that, 0 value in having a const, imo